lib/apiservers/engine/backends/container.go

```diff
@@ -2134,8 +2138,34 @@ func (c *ContainerBackend) validateContainerLogsConfig(vc *viccontainer.VicConta
 	return tailLines, since.Unix(), nil
 }
 
+func checkContainerCount(op context.Context, client *client.PortLayer) error {
+	// 0 means unlimited
+	if vchConfig.Cfg.ContainerCount == 0 {
+		return nil
+	}
+
+	all := true
+	containme, err := client.Containers.GetContainerList(containers.NewGetContainerListParamsWithContext(op).WithAll(&all))
+	if err != nil {
+		switch err := err.(type) {
+
+		case *containers.GetContainerListInternalServerError:
+			return fmt.Errorf("Error invoking GetContainerList: %s", err.Payload.Message)
+
+		default:
+			return fmt.Errorf("Error invoking GetContainerList: %s", err.Error())
+		}
+	}
+
+	count := vchConfig.Cfg.ContainerCount
+	if count == len(containme.Payload) {
```

Comment:
Any chances that `count > len(containme.Payload)` ?

Reply:
Good point. In parallel creations, there might be that case. Fixed.